### DISCUSSION TITLE
Transitland v2 Routing API: details for time window not documented?

### DISCUSSION THREAD

**Discussion by DaveNash** [transitland/transitland#434] [Q&A]
When specifying a time to the routing api (arrival or departure), what is the default window around this parameter for returned results? 
E.g., if I specify query params like '?time=09:00&arrive_by=true', what actual arrival times will be included in the returned  itineraries? Would an arrival of 08:50 be included? How about 08:45? (I presume 09:01 would not be included--?) 

Maybe more to the point: I would love min_time & max_time params so we can specify the window ourselves.

**Comment by irees** [comment]
@DaveNash The `time` value is a strict begin or end point; either itineraries starting no earlier than the provided time, or `arriveBy` itineraries that arrive at the destination stop no later than the provided time. The maximum duration of trips returned can be specified by `maxTripTime`. 

I don't think a moving window specified by min/max times would fit in well with the current algorithm, but I'll keep the request in mind.

**Reply by DaveNash** [reply]
Thanks, @irees . ...but there must be a default window 'under the hood' already, right? 
Using the example parameters I listed above, I don't see _every_ trip that arrives before 9am (i.e., no matter how early), and conversely, I don't _only_ see trips that arrive _exactly_ at 9am, so how early an arrival is included by default? From what I've seen, the call doesn't return just the closest trip (sometimes it returns more than one; other times it returns nothing, even though I know there's an arrival before the arrive_by time). So, would an arrival at 08:50 be included in the results? 08:45? 08:30? 08:00? 
Seems like it must be an at least somewhat 'fuzzy' search already; I guess I'm asking: just how fuzzy is it now? (...and can we someday specify the fuzziness?)

**Reply by DaveNash** [reply]
Hi, @irees. 
I _think_ I've encountered an example where being able to specify the allowable time window for routing requests would be very helpful. 
Using this call: https://transit.land/api/v2/routing/otp/plan?fromPlace=37.7525,-122.1989&toPlace=37.4064,-121.9668&date=2024-09-11&time=09:00&arriveBy=true&maxWalkDistance=500&maxK=2[&apikey]
to get routing from Oakland Coliseum to Great America Station, the first result actually recommends going through Great America and carrying on to Diridon, then taking VTA back to near Great America. I _presume_ this is due to the default allowable arrival time. I'd prefer to be able to specify an arrival window myself in the query params if possible. 
Thanks
